met-web/src/components/survey/create/CloneOptions.tsx

```diff
@@ -0,0 +1,232 @@
+                sort_order: sort_order,
+                search_text: search_text,
+            });
+            console.log('FETCH!');
+            console.log(fetchedEngagements);
+            setAvailableEngagements(
+                fetchedEngagements.items.filter((engagement, index) => engagement.surveys.length !== 0),
+            );
+            setLoadingEngagements(false);
+        } catch (error) {
+            dispatch(
+                openNotification({ severity: 'error', text: 'Error occurred while fetching available engagements' }),
+            );
+        }
+    };
+
+    useEffect(() => {
+        if (!availableSurveys) {
+            handleFetchSurveys(1, 10, 'asc', '');
+        }
+    }, [availableSurveys]);
+
+    useEffect(() => {
+        if (!availableEngagements) {
+            handleFetchEngagements(1, 10, 'asc', '');
+        }
+    }, [availableEngagements]);
+
+    const handleSave = async () => {
+        if (!selectedSurvey) {
+            if (selectedEngagement) return;
+            dispatch(openNotification({ severity: 'error', text: 'Please select a survey first' }));
+            return;
+        }
+        setIsSaving(true);
+        try {
+            const createdSurvey = await postSurvey({
+                name: surveyForm.name,
+                form_json: selectedEngagement ? selectedEngagement.surveys[0].form_json : selectedSurvey.form_json,
+            });
+
+            dispatch(
+                openNotification({
+                    severity: 'success',
+                    text: 'Survey created, please proceed to building it',
+                }),
+            );
+            navigate(`/surveys/${createdSurvey.id}/build`);
+        } catch (error) {
+            dispatch(
+                openNotification({
+                    severity: 'error',
+                    text: 'Error occurred while attempting to save survey',
+                }),
+            );
+            setIsSaving(false);
+        }
+    };
+
+    return (
+        <Grid container direction="row" alignItems="flex-start" justifyContent="flex-start" item xs={12} spacing={2}>
+            <Grid item xs={6}>
+                <MetLabel sx={{ marginBottom: '2px', display: 'flex' }}>
+                    Select Engagement <Typography sx={{ ml: 1, color: '#ACA9A9' }}>(optional)</Typography>
+                </MetLabel>
+
+                <Autocomplete
+                    id="engagement-selector"
+                    options={availableEngagements || []}
+                    renderInput={(params) => (
+                        <TextField
+                            {...params}
+                            label=" "
+                            InputLabelProps={{
+                                shrink: false,
+                            }}
+                            fullWidth
+                            onChange={(e) => {
+                                handleFetchEngagements(1, 10, 'asc', e.target.value);
```

Comment:
1,10 i think are the page size rite? can you please use a constant here?
https://refactoring.guru/replace-magic-number-with-symbolic-constant

Reply:
smart